docs/contributor/03-11-hap-rules.md

```diff
@@ -0,0 +1,285 @@
+
+SecretBinding pools
+- hyperscalerType: gcp_us-central1, 
+```
+
+### Shared & EU Access attribute
+
+The `shared` and `euAccess` attributes does not correspond to any physical SKR property. It is a logical attribute that is used only to add its label. If the rule entry contains `shared` or `euAccess` attributes then when triggered a `shared: true` or `euAccess: true` labels are added to the [search labels](#search-labels) depending on which one of those two attributes is listed in the rule entry. Shared label on a SecretBindings marked it as assignable to more than one SKR and euAccess is a non-functional label not reused by KEB (more on that can be found in p[HAP Pool document](03-10-hyperscaler-account-pool.md)). Below configuration specifies that all gcp clusters will use the same pool of shared secret bindings marked with labels `hyperscalerType: gcp`, `shared: true` and azure clusters in the region cf-ch20 will use the pool of secret bindings marked with labels `hyperscalerType: azure`, `euAccess: true`. 
+
+```
+hap: 
+  rule: 
+    - gcp -> S
+    - azure(PR=cf-ch20) -> E
+
+SecretBinding pools:
+- hyperscalerType: gcp; shared: true
+- hyperscalerType: azure_cf-ch20, euAccess: true
+```
+
+### Attribute with "*"
+
+Input attributes values can be replaced with `*` which means that all SKR values of that attribute are matched and reused in unchanged form when applying [search labels](#search-labels) modifications. For example, the following configuration makes KEB search SecretBindings with `hyperscalerType: gcp_<PR>`, where `PR` corresponds to an SKR value, for all `gcp` clusters:
+
+```
+hap: 
+  rule: 
+  - gcp(PR=*)
+```
+
+```
+SecretBinding pools:
+- hyperscalerType: gcp_cf-sa30
+- hyperscalerType: gcp_cf-jp30
+```
+
+The attributes that support `*` include: `PR`, `CR`.
+
+> ![NOTE]
+> The above configuration example effectively disables usage of SecretBindings with only `hyperscalerType: gcp` label.
+
+### Atributes Summary
+
+The following table summarizes the attributes that can be used in the rule entry. Its columns define:
+* Name and symbol - the later is used in rule entries.
+* Data type and possible values.
+* Input attributes - if value is true then the attribute can occure in the input attributes section (left part) of a rule entry (see [Rule Format](#rule-format) section).
+* Output attributes - if value is true then the attribute can occure in the output attributes section (right part) of a rule entry (see [Rule Format](#rule-format) section).
+* Modified SecretBinding search labels - lists modifications of search labels that the rule entry containing the attribute adds to search labels if triggered. 
+
+| Name (Symbol)        	| Data Type and Possible Values                                                                                                                    	| Input Attribute 	| Output Attribute 	| Modified SecretBinding Search Labels                                                              	|
```

Comment:
PR and CR are as well output params. Add another column with location (right left) then:
in out location
1) PR is true true left
2) S is false true right 
etc

Reply:
Is it not excessive @PK85 ? I thought Input, Output and Search Labels are enough to define all we need.

Reply:
I think such column is redundant because input parameters are on the left of the rule and output on the right.

Reply:
> PR and CR are as well output params. Add another column with location (right left) then: in out location
> 
> 1. PR is true true left
> 2. S is false true right
>    etc

We have discrepancy.
Are PR and HR output parameters? According to the table not.
There are not on the right side of the rule (after the `->`).
But they are used in hyperscaler type.